### PR TITLE
fix: remark-lint, caches and preset-lint-node

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR is pretty much (is a second attempt) at working on some of our `remark-lint` rules. (Based on the prior merged PR (https://github.com/nodejs/nodejs.org/pull/5999), I've realized that not adopting `remark-preset-lint-node` as a "compatibility issue" is an excuse.

I've read a few docs and tried to play around with a few things.

- First, we shouldn't cache lint rules with Turborepo. Even more, as this repository is ever-changing, it only feels that having a separate file, `turbo.json`, to keep track of every single file does feel like something at best flaky and waiting to break apart.
- Going through Node.js's Website repository history, I've found a set of Remark settings on `.remarkc` that actually explain the misformatting of Frontmatter when `eslint-plugin-mdx` was attempting to run Remark Lint on files. Apparently, just by removing `remark-lint-preset-node`, the issue itself wouldn't have been fixed.
- I've found that we can actually adopt the `remark-preset-lint-node` with a few tweaks, such as manually disabling `remark-gfm` (as we don't need it) (and apparently on the repository history many eons ago it was at some point disabled). (History often taught us lessons!)
- I've found that even tho locally `eslint` and `prettier` caching with `metadata` approach is way faster, apparently on CI it isn't (and actually completely ineffective) due to Git of course not storing timestamp metadata 🤦 (my bad!)
  - Switching to content checks should be a little bit slower, but still way faster than current times on CI!

This PR in the end ships some CI speed improvements for caching, Better Remark Linting configuration, and enabled Stylelint to also use its cache feature.

## Validation

Linting should pass as usual (Note that some files got updated).

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
I want to dig a bit further into https://github.com/nodejs/remark-preset-lint-node before weighing in

File: .gitignore [hunk 1]
```diff
@@ -3,11 +3,6 @@ node_modules
 npm-debug.log
 .npm
 
-# OSX system files, the bane of our existence
-.DS_Store
-.AppleDouble
-.LSOverride
```
Comment (bmuenzenmeyer) [inline .gitignore:9]: are we expecting users to understand to ignore these at the user/global level? hoping it doesn't come back to cause trouble

File: .remarkrc.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "plugins": [
+    "remark-preset-prettier",
+    "remark-frontmatter",
+    "remark-preset-lint-node",
+    ["remark-gfm", false],
```
Comment (bmuenzenmeyer) [inline .remarkrc.json:6]: TIL you can turn off an inherited plugin like this

File: turbo.json [hunk 3]
```diff
@@ -49,123 +67,49 @@
...
-        ".eslintignore",
-        ".stylelintrc.mjs",
-        ".stylelintignore",
-        ".prettierrc.json",
-        ".prettierignore",
-        ".remarkrc.json",
-        ".lintstaged.json"
-      ],
+      "cache": false,
+      "persistent": true,
```
Comment (bmuenzenmeyer) [inline turbo.json]: I don't think `persistent` necessary here and in other `:fix` scenarios. It was my understanding that `persistent` was used for commands that watch by default and these commands do not.

https://turbo.build/repo/docs/reference/configuration#persistent

**Review by bmuenzenmeyer — CHANGES_REQUESTED** [review]
small question about the `persist` keys the in `turbo.json`

**Comment by ovflowd** [comment]
> I want to dig a bit further into [nodejs/remark-preset-lint-node](https://github.com/nodejs/remark-preset-lint-node) before weighing in

Right, take your time :)

File: .remarkrc.json (hunk 2 above)
Comment (ovflowd) [reply .remarkrc.json:6]: Same, only learned it now!

File: turbo.json (hunk 3 above)
Comment (ovflowd) [reply turbo.json]: Oh! TIL I learned that! Will make proper updates!

File: .gitignore (hunk 1 above)
Comment (ovflowd) [reply .gitignore:9]: Right, I feel that we should avoid having env specific / platform specific ignores. And my experience in general is that these hardly ever get committed. And if ever, they can just delete them.

**Review by bmuenzenmeyer — APPROVED** [review]
I appreciate your commitment and drive to get this right, even if we don't always know what that is in the moment. Progress over perfection, and humility to go back and fix things 😄

**Review by sheplu — COMMENTED** [review]
LGTM

**Review by sheplu — APPROVED** [review]
LGTM

**Pull request merged**